mindmeld/models/embedder_models.py

```diff
@@ -14,119 +14,731 @@
 """
 This module contains the embedder model class.
 """
-from abc import ABC, abstractmethod
-import pickle
+import json
 import logging
 import os
+import pickle
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+
 import numpy as np
+from tqdm.autonotebook import trange
 
-from .. import path
+from ._util import _is_module_available, _get_module_or_attr as _getattr
 from .helpers import register_embedder
 from .taggers.embeddings import WordSequenceEmbedding
+from .. import path
+from ..core import Bunch
+from ..resource_loader import Hasher
 from ..tokenizer import Tokenizer
 
-
 logger = logging.getLogger(__name__)
 
-try:
-    from sentence_transformers import SentenceTransformer
 
-    register_bert = True
-except ImportError:
-    logger.info("Must install the extra [bert] to use the built in embbedder.")
-    register_bert = False
+def _torch(op, *args, sub="", **kwargs):
+    return _getattr(f"torch{'.' + sub if sub else ''}", op)(*args, **kwargs)
 
 
 class Embedder(ABC):
     """
     Base class for embedder model
     """
 
-    def __init__(self, app_path, **kwargs):
-        """Initializes an embedder."""
-        self.model_name = kwargs.get("model_name", "default")
-        self.cache_path = path.get_embedder_cache_file_path(
-            app_path, kwargs.get("embedder_type", "default"), self.model_name
-        )
+    class EmbeddingsCache:
+
+        def __init__(self, cache_path):
+            self.cache_path = cache_path
+            self.reset_data()
+            self.load()
+
+        def reset_data(self):
+            self.data = OrderedDict()
+
+        def load(self):
+            """Loads the cache file."""
+
+            if os.path.exists(self.cache_path) and os.path.getsize(self.cache_path) > 0:
+                with open(self.cache_path, "rb") as fp:
+                    data = pickle.load(fp)
+
+                if (
+                    "_texts" in data and
+                    "_texts_embeddings" in data and
+                    isinstance(data["_texts"], list)
+                ):  # new format;
+                    self.data = dict(zip(data["_texts"], data["_texts_embeddings"]))
+
+                elif (
+                    "synonyms" in data and
+                    "synonyms_embs" in data and
+                    isinstance(data["synonyms"], dict)
+                ):  # deprecated format; backwards compatible with ER module code
+                    self.data = {key: data["synonyms_embs"][j] for key, j in data["synonyms"]}
+
+                else:  # deprecated format; backwards compatible with QA module code
+                    if not isinstance(data, dict):
+                        msg = "Unknown data format while loading cache embeddings. " \
+                              "Ignoring loading ..."
+                        logger.error(msg)
+                    self.data = data
```

Comment:
So because this `load()` method belongs to a cache class, i think it would make sense to call a load and expect that a cache file is loaded into memory; and for that reason we assign here.